homeassistant/components/reolink/__init__.py

```diff
@@ -175,3 +178,56 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         hass.data[DOMAIN].pop(config_entry.entry_id)
 
     return unload_ok
+
+
+def cleanup_disconnected_cams(
+    hass: HomeAssistant, config_entry_id: str, host: ReolinkHost
+) -> None:
+    """Clean-up disconnected camera channels or channels where a different model camera is connected."""
+    if not host.api.is_nvr:
+        return
+
+    device_reg = dr.async_get(hass)
+    devices = dr.async_entries_for_config_entry(device_reg, config_entry_id)
+    for device in devices:
+        device_id = [
+            dev_id[1].split("_ch")
+            for dev_id in device.identifiers
+            if dev_id[0] == DOMAIN
+        ][0]
+
+        if len(device_id) < 2:
+            # Do not consider the NVR itself
+            continue
+
+        ch = int(device_id[1])
+        ch_model = host.api.camera_model(ch)
+        remove = False
+        if ch not in host.api.channels:
+            remove = True
+            _LOGGER.info(
+                "Removing Reolink device %s, since no camera is connected to NVR channel %s anymore",
+                device.name,
+                ch,
+            )
```

Comment:
This is a normal operation; I think this should be demoted to debug.

```suggestion
            _LOGGER.debug(
                "Removing Reolink device %s, since no camera is connected to NVR channel %s anymore",
                device.name,
                ch,
            )
```